### PR TITLE
LoRa Modulation

### PR DESCRIPTION
**Pull request by lthiery**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,8 @@ description = "A LoRa physical layer implementation enabling utilization of a ra
 [dependencies]
 defmt = { version = "0.3" }
 log = { version = "0.4" }
+#lora-modulation = { version = "0.1.1" }
+lora-modulation = { git = "https://github.com/ivajloip/rust-lorawan.git", branch = "lthiery/impl-sfbw" }
```
Comment (lucasgranberg) [inline Cargo.toml]: @ceekdee will have to decide on this but it feels like this crate should not depend on a lorawan crate. For p2p purposes this requirement is a bit backward.

Comment (lthiery) [reply Cargo.toml]: technically, the crate is "lora-modulation" and happens to be hosted in the rust-lorawan repo. As you can see above, the rust-lorawan repo isn't in the name when you pull it from crates.io

Comment (ceekdee) [reply Cargo.toml]: The lora-modulation crate will be used by lora-phy version 2 once it is in final form on crates.io.

**Comment by lthiery** [comment]
So content is otherwise okay @lucasgranberg and @ceekdee ?

Once I can push lora-modulation to crates.io with a new version number, we can depend on lora-modulation 0.1.1 here.

**Comment by ceekdee** [comment]
If I understand the intent correctly, lora-phy version 2 will depend on lora-modulation for its SF, BW, and CR values.  Therefore, I assume:

https://github.com/ivajloip/rust-lorawan/blob/master/modulation/Cargo.toml will have the dependency on lora-phy removed
https://github.com/ivajloip/rust-lorawan/blob/master/modulation/src/lora_phy.rs will be removed

before lora-modulation is pushed to crates.io at 0.1.1?  Or am I looking at it incorrectly?

**Comment by lthiery** [comment]
Yes, although I don't think the lora-phy dependency _needs_ removing before pushing 0.1.1 since it's behind a feature-flag? In other words, lora-phy will depend on lora-modulation with `external-lora-phy` disabled and there's no concern about circular dependency.

This keeps things working during the transition. Once lora-phy adopts lora-modulation, lora-modulation should deprecate the feature in future releases, but I don't think there's an urgency in creating a new release for that necessarily?

**Comment by ceekdee** [comment]
Understood.

**Comment by lthiery** [comment]
Things is good to go now. When there's a new release tagged for lora-phy, I'll make the updates in rust-lorawan.

**Review by ceekdee — APPROVED** [review]

**Comment by ceekdee** [comment]
@lthiery : once the issue I communicated in our chat is addressed in lora-modulations (please correct me if different changes are needed) and a new release of lora-modulations is created on crates.io, I will update lora-phy version 2 to use that version and update embedded-hal-async to the latest version before creating a new lora-phy version 2 on crates.io.

**Pull request merged**